Gems/Atom/RHI/Vulkan/Code/Source/RHI/StreamingImagePool.cpp

```diff
@@ -42,18 +117,26 @@ namespace AZ
             }
            
             RHI::HeapMemoryLevel heapMemoryLevel = RHI::HeapMemoryLevel::Device;
-            m_memoryAllocatorUsage = m_memoryUsage.GetHeapMemoryUsage(heapMemoryLevel);
+            RHI::HeapMemoryUsage& heapMemoryUsage = m_memoryUsage.GetHeapMemoryUsage(heapMemoryLevel);
             MemoryAllocator::Descriptor memoryAllocDescriptor;
             memoryAllocDescriptor.m_device = &device;
             memoryAllocDescriptor.m_pageSizeInBytes = RHI::RHISystemInterface::Get()->GetPlatformLimitsDescriptor()->m_platformDefaultValues.m_imagePoolPageSizeInBytes;
             memoryAllocDescriptor.m_heapMemoryLevel = heapMemoryLevel;
             memoryAllocDescriptor.m_memoryTypeBits = memoryTypeBits;
-            // We don't pass the ResourcePool MemoryUsage because we will manually control residency when expanding/trimming.
-            memoryAllocDescriptor.m_getHeapMemoryUsageFunction = [this]() { return &m_memoryAllocatorUsage; };
-            memoryAllocDescriptor.m_recycleOnCollect = true;
-            memoryAllocDescriptor.m_collectLatency = RHI::Limits::Device::FrameCountMax;
+            memoryAllocDescriptor.m_getHeapMemoryUsageFunction = [&]() { return &heapMemoryUsage; };
+            memoryAllocDescriptor.m_recycleOnCollect = false;
+            memoryAllocDescriptor.m_collectLatency = 0;
             m_memoryAllocator.Init(memoryAllocDescriptor);
 
+            // Initialize tile allocator
+            m_heapPageAllocator.Init(memoryAllocDescriptor);
+            const uint32_t TileSizeInBytes = SparseImageInfo::StandardBlockSize;
+            TileAllocator::Descriptor tileAllocatorDesc;
+            tileAllocatorDesc.m_tileSizeInBytes = TileSizeInBytes;
+            // Tile allocator updates used resident memory
+            tileAllocatorDesc.m_getHeapMemoryUsageFunction = [&]() { return &heapMemoryUsage; };
```

Comment:
Just curious, is there a reason we don't just store a pointer to the heap memory usage object on the allocator desc and use the callback instead?

Reply:
A pointer should be good. The getXxx function was from previous allocator code pattern which I didn't pay much attention to. This will be changed.